packages/netlify-cms-backend-github/src/implementation.js

```diff
@@ -484,10 +501,9 @@ export default class GitHub {
 
   publishUnpublishedEntry(collection, slug) {
     // publishUnpublishedEntry is a transactional operation
-    return this.runWithLock(async () => {
-      const metaData = await this.api.publishUnpublishedEntry(collection, slug);
-      const mediaFiles = await this.getMediaFiles({ metaData });
-      return { mediaFiles };
-    }, 'Failed to acquire publish entry lock');
+    return this.runWithLock(
```

Comment:
Instead of returning the list of files, the UI will reload the media library after publish